### ISSUE TITLE
Vision Dataset Proposals

### ISSUE THREAD

**Issue by atharva-tendle** [ManifoldRG/NEKO_Archive#5]
**Goal:**

We want to look into the vision datasets used in the GATO paper as a starting point and survey for information related to those.
Ideally, we would like to know the following pieces of information for each dataset:
- Is it open source?
- Does it have proper licensing?
- Can we easily use it for training (code availability etc.)

More items can be added to the list above but these are just some rough ideas.

**Output:** 

- Update the dataset Google sheet with the above information. 
- For the datasets that are not easily accessible, we might want to research alternatives and do that in a systematic fashion by understanding the desirable characteristics of a replacement.

**Comment by harshsikka** [comment]
@atharva-tendle reviewing this again, would be good to reach out to Helen. I don't think she's part of the GH organization yet, but perhaps on discord. She said she wanted to think through what kind of information quality makes a "good dataset" for our purposes, and to have a basic criteria to rate datasets as we build this mage one out.

**Comment by atharva-tendle** [comment]
Yep! will do that. I was going to put together some basic information. I can ping her once I have this.

**Comment by harshsikka** [comment]
@helenlu66 is here!

**Comment by harshsikka** [comment]
@atharva-tendle checkout my comment [here](https://github.com/ManifoldRG/NEKO/issues/4#issuecomment-1554636373).

If we can get a summary by the group meeting on 5/21/2023, of vision, language ( #6 ), and control ( #4 ), and some candidate objective combinations, we are in a good spot to scope out the mid term experimental aims and the long term as well.

**Comment by atharva-tendle** [comment]
# Recommendation and Plan

## Status
Currently, only the following datasets are publicly available to use (there are all vision + language datasets)

- MS COCO Captions
- Conceptual Captions
- QKVQA
- VQAv2

## Recommendation
Until we have an embedding function that works for different modalities we should focus on one of these datasets.
I would propose using the MS COCO Captions as it comes with PyTorch and would require minimal effort to load into the code.

In parallel, I think there are two main threads to focus on:

- Code for processing other datasets that are open source.
- Individual tickets for digging into alternatives for the remaining datasets listed in this [sheet](https://docs.google.com/spreadsheets/d/18t38QKIMsZXiNxRcgDdKIxcrclPb5QEzvVKvGQBSJ98/edit#gid=1513169105).

## Plan

1. Start building the embedding function and the encoder and test it out with MS COCO. 

> - NOTE: Embedding would need to be tested with some RL dataset + MS COCO.

2. Create another ticket for creating loaders for the remaining open-source datasets.
> - These can be individual tickets but I would propose a long ticket handled by one person so that the code is written in a way that doesn't need future refactoring (proper data classes etc.)

3. Create tickets for the MassiveText, ALIGN, M3W, and LTIP datasets as we need to find good alternatives for each.
> - I would propose individual tickets for each since it would require some digging into what is a good alternative and scoping the cost for processing this data.

**Comment by atharva-tendle** [comment]
Closing Issue. Created scoped-out follow-up issues in the backlog.
